### ISSUE TITLE
Add an ID to the Integration class to distinguish between flows in the UI

### ISSUE THREAD

**Issue by lordrip** [KaotoIO/kaoto-backend#663]
Currently, there's no flow ID in [the Integration class](https://github.com/KaotoIO/kaoto-backend/blob/f76047bf4ce30c8585dd3da8706f944dfd975dbc/api/src/main/java/io/kaoto/backend/api/resource/v1/model/Integration.java), this means that whenever we sync code from the source code editor, the UI doesn't have a way to distinguish between flows, and therefore, the steps from each flow don't have ID either.

The goal would be for the backend to include a new ID property in the Integration class to distinguish between different flows. If the UI already provides an ID, the backend will use it, otherwise, the backend will create one. This is not relevant at the moment for the YAML code since this new ID will live in the communication channel between the backend - frontend.

In summary:

* UI to Backend -> We send the ID if available, then the backend will keep it
* Backend to UI -> Based on what's available, being an existing ID, the current DSL, a timestamp, or `metadata.name`, it sends it back to the UI

**Comment by lordrip** [comment]
Related issue: https://github.com/KaotoIO/kaoto-ui/issues/1910

**Comment by Delawen** [comment]
What we are going to use as id is the `name`. But we have to make sure the `name` is always unique inside the same set of flows.
